homeassistant/components/shelly/config_flow.py

```diff
@@ -460,7 +487,7 @@ def async_supports_options_flow(cls, config_entry: ConfigEntry) -> bool:
         return (
             get_device_entry_gen(config_entry) in RPC_GENERATIONS
             and not config_entry.data.get(CONF_SLEEP_PERIOD)
-            and config_entry.data.get("model") != MODEL_WALL_DISPLAY
+            and config_entry.data.get(CONF_SCRIPT, True)
```

Comment:
Looks like this code is similar to what appears in __init__. Can it be a utility function?

Reply:
Yup, done